### PR TITLE
Forum Topic Page Improvement -> Question, Label, and Vote

### PR DESCRIPTION
**Pull request by bivanalhar**
Resolved issue https://github.com/Coursemology/coursemology2/issues/7439, https://github.com/Coursemology/coursemology2/issues/7440, and https://github.com/Coursemology/coursemology2/issues/7441

### CODE REVIEW

File: client/app/bundles/course/forum/components/buttons/HideButton.tsx [hunk 1]
```diff
@@ -18,6 +18,14 @@ const translations = defineMessages({
     id: 'course.forum.HideButton.unhide',
     defaultMessage: 'Unhide',
   },
+  hideTag: {
```
Comment (cysjonathan) [inline client/app/bundles/course/forum/components/buttons/HideButton.tsx:21]: What about translation files?

File: app/controllers/course/forum/posts_controller.rb [hunk 2]
```diff
@@ -51,7 +51,14 @@ def destroy
...
       @topic.update_column(:latest_post_at, @topic.posts.last&.created_at || @topic.created_at)
-      render json: { topicId: @topic.id, postTreeIds: @topic.posts.ordered_topologically.sorted_ids }
+      if @topic.topic_type == 'question'
+        @topic.specific.update_resolve_status
+        render json: { topicId: @topic.id,
+                       postTreeIds: @topic.posts.ordered_topologically.sorted_ids,
+                       isTopicResolved: @topic.reload.resolved? }
+      else
+        render json: { topicId: @topic.id, postTreeIds: @topic.posts.ordered_topologically.sorted_ids }
+      end
```
Comment (cysjonathan) [inline app/controllers/course/forum/posts_controller.rb:61]: Your code is duplicating the render code in line 56 and line 60.  
You should do it this way:

if topic is question
`-> update resolve status
render code

Then you only have 1 render code.

Also, you only need to update resolve status iff topic is question AND post is marked as answer.

File: client/app/bundles/course/forum/components/buttons/VotePostButton.tsx [hunk 3]
```diff
@@ -47,15 +47,15 @@ const VotePostButton: FC<Props> = ({ post }) => {
     return (
       <div className="flex items-center">
         <IconButton
-          color="info"
+          color="success"
           onClick={(): void => handleVotePost(1)}
           title="Upvote"
         >
-          <ThumbUpOffAlt />
+          <ThumbUpOffAlt className="fill-green-600" />
```
Comment (cysjonathan) [inline client/app/bundles/course/forum/components/buttons/VotePostButton.tsx:54]: Why does ThumbUpOffAlt need to fill-green-600 but ThumbDownOffAlt not need any className?

File: client/app/bundles/course/forum/components/buttons/HideButton.tsx (hunk 1 above)
Comment (bivanalhar) [reply client/app/bundles/course/forum/components/buttons/HideButton.tsx:21]: updated.
Also, I have updated the mandarin one, though you might need to double check if my translation is correct already or not

File: client/app/bundles/course/forum/components/buttons/VotePostButton.tsx (hunk 3 above)
Comment (bivanalhar) [reply client/app/bundles/course/forum/components/buttons/VotePostButton.tsx]: sorry it should not have been there.
I checked our codebase, and apparently for this kind of icon, most places use `IconButton` so I preserve it as it is, merely changing the color to become "success" for ThumbsUp and "error" for ThumbsDown

File: app/controllers/course/forum/posts_controller.rb (hunk 2 above)
Comment (bivanalhar) [reply app/controllers/course/forum/posts_controller.rb]: Roger that, and done.

**Pull request merged**